triedb/hashdb/database.go

```diff
@@ -137,6 +138,8 @@ type Database struct {
 	childrenSize common.StorageSize // Storage size of the external children tracking
 
 	lock sync.RWMutex
+
+	referenceRoot bool
```

Comment:
(optional) Consider making this `referenceRootOnUpdate`. It's a little less important IMO than the exported one in `Config` as it's very close to where it's used, but still out of context. I don't think you need to call it "atomic" here because that's a given as there are locks everywhere in the code.